### DISCUSSION TITLE
antd5在next.js13中遇到的问题

### DISCUSSION THREAD

**Discussion by Tongyaoyu** [ant-design/ant-design#41498] [Q&A]
1.在next.js13中使用antd5，刷新时页面会没有样式地闪一下，然后恢复样式
2.在next.js13中想覆盖antd5的默认样式，例如想在css module中覆盖菜单栏的样式，由于css module会给类加哈希值，所以使用global包含类名就不会产生哈希值，并变为全局样式，代码如下
```
.menu_item :global(.ant-menu-item-selected) {
    background-color: #008080 !important;
}
```
但是并不起效果，换其他antd中的类名也是不起效果，不知道该怎么解决

**Comment by Tongyaoyu** [comment]
有大佬可以解答一下吗

**Comment by arvinxx** [comment]
> 1.在next.js13中使用antd5，刷新时页面会没有样式地闪一下，然后恢复样式

这个是因为 SSR 丢样式，文档里去找下SSR集成的部分就好

> 2. 但是并不起效果，换其他antd中的类名也是不起效果，不知道该怎么解决

给个复现

**Reply by Tongyaoyu** [reply]
使用antd5中的菜单组件，代码如下，menu和menu_item是自己加的类名
```
<Menu className={style.menu} defaultSelectedKeys={['1']} mode="inline">
                  {items.map((menu) => (
                    <Menu.Item key={menu.key} icon={menu.icon} className={style.menu_item} >
                      {menu.label}
                    </Menu.Item>
                  ))}
              </Menu>
```
假如想改变菜单项被选中时的背景颜色，antd5中的类名是ant-menu-item-selected，在相应的css module中代码如下
```
.menu_item :global(.ant-menu-item-selected) {
    background-color: #008080 !important;
}
```
必须要加自己的类名，否则会报错：Selector ":global(.ant-menu-item-selected)" is not pure (pure selectors must contain at least one local class or id) ，加上就不会报错，但是页面不起作用，还是默认的样式。按道理说global已经不会让ant-menu-item-selected加哈希值了，不知道为啥还是不起作用

**Reply by MadCcc** [reply]
Nextjs 不能在 root 使用 `:global`，加上自己类名后请检查 selector 是否对的上层级

**Reply by Tongyaoyu** [reply]
谢谢解答，确实是层级不对，这么写就对了
```css
.menu :global .ant-menu-item.ant-menu-item-selected {
    background-color: red ;
}
```
